tests/integration/app-charm/src/charm.py

```diff
@@ -244,17 +260,40 @@ def create_certificate(self, event: ActionEvent):
     def run_mtls_producer(self, event: ActionEvent):
         broker_ca = event.params["broker-ca"]
         bootstrap_server = event.params["bootstrap-server"]
+        num_messages = int(event.params["num-messages"])
+
+        decode_cert = base64.b64decode(broker_ca).decode("utf-8").strip()
 
         try:
-            self._create_truststore(broker_ca=broker_ca)
-            self._attempt_mtls_connection(bootstrap_servers=bootstrap_server)
+            self._create_truststore(broker_ca=decode_cert)
+            self._attempt_mtls_connection(
+                bootstrap_servers=bootstrap_server, num_messages=num_messages
+            )
         except Exception as e:
             logger.error(e)
             event.fail()
             return
 
         event.set_results({"success": "TRUE"})
```

Comment:
**question:** Was this not raising? If it isn't raising, do we know why?

Reply:
I believe that line `self._attempt_mtls_connection` is not raising any Exception, even when the `subprocess` call fails. Therefore, even when no data are written to kafka, this action would not fail